dev_guide/templates.adoc

```diff
@@ -50,20 +55,54 @@ set of labels to apply to every resource created by the template.
 
 <1> The name of the template
 <2> Optional description for the template
-<3> A list of parameters for the template
-<4> A list of labels to apply to resources
-<5> A list of resources to create
+<3> The icon that will be shown in the UI for this template
+<4> A list of tags that this template will have in the UI
+<5> A list of parameters for the template
+<6> A list of labels to apply to all resources
+<7> A list of resources to create
 
 ====
 
-== Creating a Configuration From a Template
-You can create a configuration from a template using the CLI or, if a template
+== Creating a list of resources from a Template
+You can create a list of resources from a template using the CLI or, if a template
 has been uploaded to your project or global template library, using the web
 console.
 
 Create a template JSON file, like the above example, then upload it with the
 CLI using the following process:
 
+=== Defining Template parameters
+
+Each parameter describe a variable and the variable value which can be
+referenced in any text field in the `objects` list field. The value can be set
+explicitly or it can be generated by the OpenShift.
+
+You can assign an explicit value to the parameter use the `value` field:
+
+----
+"parameters": [
+  {
+    "name": "USERNAME",
+    "description": "The user name for Joe",
+    "value": "joe"
+  }
+],
+----
+
+To generate a value using pseudo-regexp syntax you have to specify the
+`generate` field and the `from` field:
+
+----
+"parameters": [
+  {
+    "name": "PASSWORD",
+    "description": "The random user password",
+    "generate": "expression",
```

Comment:
@soltysh the `generate` will always be set to `expression` as we don't have any other generators (and most like won't add new one before 3.0.1).... Instead of footnotes I would rather explain the value in the text above... to make it consistent with other examples.